liquidctl/driver/krakenz3.py

```diff
@@ -0,0 +1,676 @@
+    "backwards-marquee-4": (0x03, 0x04, 2, 1, 1),
+    "backwards-marquee-5": (0x03, 0x05, 2, 1, 1),
+    "backwards-marquee-6": (0x03, 0x06, 2, 1, 1),
+    "covering-backwards-marquee": (0x04, 0x00, 2, 1, 8),
+    "backwards-moving-alternating-3": (0x05, 0x03, 4, 1, 2),
+    "backwards-moving-alternating-4": (0x05, 0x04, 4, 1, 2),
+    "backwards-moving-alternating-5": (0x05, 0x05, 4, 1, 2),
+    "backwards-moving-alternating-6": (0x05, 0x06, 4, 1, 2),
+    "backwards-rainbow-flow": (0x0B, 0x00, 2, 0, 0),
+    "backwards-super-rainbow": (0x0C, 0x00, 2, 0, 0),
+    "backwards-rainbow-pulse": (0x0D, 0x00, 2, 0, 0),
+}
+
+# A static value per channel that is somehow related to animation time and
+# synchronization, although the specific mechanism is not yet understood.
+# Could require information from `initialize`, but more testing is required.
+_STATIC_VALUE = {
+    0b001: 40,  # may result in long all-off intervals (FIXME?)
+    0b010: 8,
+    0b100: 1,
+    0b111: 40,  # may result in long all-off intervals (FIXME?)
+}
+
+# Speed scale/timing bytes
+# scale -> (slowest, slower, normal, faster, fastest)
+_SPEED_VALUE = {
+    0: ([0x32, 0x00], [0x32, 0x00], [0x32, 0x00], [0x32, 0x00], [0x32, 0x00]),
+    1: ([0x50, 0x00], [0x3C, 0x00], [0x28, 0x00], [0x14, 0x00], [0x0A, 0x00]),
+    2: ([0x5E, 0x01], [0x2C, 0x01], [0xFA, 0x00], [0x96, 0x00], [0x50, 0x00]),
+    3: ([0x40, 0x06], [0x14, 0x05], [0xE8, 0x03], [0x20, 0x03], [0x58, 0x02]),
+    4: ([0x20, 0x03], [0xBC, 0x02], [0xF4, 0x01], [0x90, 0x01], [0x2C, 0x01]),
+    5: ([0x19, 0x00], [0x14, 0x00], [0x0F, 0x00], [0x07, 0x00], [0x04, 0x00]),
+    6: ([0x28, 0x00], [0x1E, 0x00], [0x14, 0x00], [0x0A, 0x00], [0x04, 0x00]),
+    7: ([0x32, 0x00], [0x28, 0x00], [0x1E, 0x00], [0x14, 0x00], [0x0A, 0x00]),
+    8: ([0x14, 0x00], [0x14, 0x00], [0x14, 0x00], [0x14, 0x00], [0x14, 0x00]),
+    9: ([0x00, 0x00], [0x00, 0x00], [0x00, 0x00], [0x00, 0x00], [0x00, 0x00]),
+    10: ([0x37, 0x00], [0x28, 0x00], [0x19, 0x00], [0x0A, 0x00], [0x00, 0x00]),
+    11: ([0x6E, 0x00], [0x53, 0x00], [0x39, 0x00], [0x2E, 0x00], [0x20, 0x00]),
+}
+
+_ANIMATION_SPEEDS = {
+    "slowest": 0x0,
+    "slower": 0x1,
+    "normal": 0x2,
+    "faster": 0x3,
+    "fastest": 0x4,
+}
+
+
+class KrakenZ3(UsbDriver):
+    """Fourth-generation Kraken Z liquid cooler."""
+
+    SUPPORTED_DEVICES = [
+        (
+            0x1E71,
+            0x3008,
+            None,
+            "NZXT Kraken Z (Z53, Z63 or Z73)",
+            {
+                "speed_channels": _SPEED_CHANNELS_KRAKENZ,
+                "color_channels": _COLOR_CHANNELS_KRAKENZ,
+            },
+        )
+    ]
+
+    def __init__(self, device, description, speed_channels, color_channels, **kwargs):
+        super().__init__(device, description, **kwargs)
+
+        hidinfo = next(
+            info
+            for info in hid.enumerate(device.vendor_id, device.product_id)
+            if info["serial_number"] == device.serial_number
```

Comment:
I have been told that the Kraken Z serial number descriptor is not a true serial number, but just some constant string (probably related to hardware or shipped-firmware version).

Can you show me what the serial number for your device(s) looks like?